extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestRowGroupInfoVisitor.java

```diff
@@ -0,0 +1,277 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table;
+
+import io.deephaven.engine.table.Table;
+import io.deephaven.parquet.table.metadata.RowGroupInfo;
+import io.deephaven.util.function.ThrowingConsumer;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.JsonNode;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static java.util.Map.entry;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+/**
+ * Test serialization/deserialization of {@link RowGroupInfo} implementations. This test is intentionally in a different
+ * package than {@link RowGroupInfo} so that we're able to verify we have access (not private, not package private) to
+ * each method we would need for serialization/deserialization
+ */
+public class TestRowGroupInfoVisitor {
```

Comment:
This seems like a very JSON-specific test that could be done in a simpler fashion - simply, making sure you can read the values you are expecting to be able to read. In this regard, I think just doing what you are doing in `RowGroup2SelfVisitor` is better without adding the complication of JSON?